### PR TITLE
Add client metrics to rest client extension

### LINKED ISSUE #13847: Rest-client extension does not record any metrics about client calls
I followed [the official documentation](https://quarkus.io/guides/rest-client) to create an app that uses the rest-client extension to make http calls.
I also added the micrometer lib to the project, so I get metrics.
My expectation is that I can get out-of-the-box metrics about rate, latency (percentiles), errors, etc on the clients calls.

But when I execute my example (and call localhost:8080/country/name/{name}), I do not see any client related metrics, even though I can see the result of the client call on my browser.

# Steps to reproduce:
1. Clone https://github.com/oscarfh/quarkus-rest-metrics
2. Run ./mvnw compile quarkus:dev
3. go to http://localhost:8080/country/name/germany
4. go to http://0.0.0.0:8080/metrics
5. Notice that there is no metric about the rest client call performed when you executed item 3

### PR DESCRIPTION
**Pull request by oscarfh**
This is my take (with a lot of support from @ebullient ) on how to add metrics to the rest-clients automatically (#13884 )

Please let me know of the improvements, or also feel free to make it yourselves.

Resolves #13847

### CODE REVIEW

**Comment by ebullient** [comment]
I'd like to see if we can streamline things.. but this is a good start. Catch you on Zulip. ;)

**Review by gsmet — DISMISSED** [review]
Thanks! I will let @ebullient drive this, just a note that this needs to be squashed before merging. Please dismiss when done!

**Review by ebullient — CHANGES_REQUESTED** [review]
I've made a PR against the original branch w/ updates to move the http client metrics extension closer to http server metrics collection (as reactive rest is going to influence both paths across multiple extensions.. )

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 1]
```diff
@@ -70,6 +70,8 @@
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.metrics.MetricsCapabilityBuildItem;
```
Comment (kenfinnigan) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:73]: Imports added that don't appear to be used, is that right?

File: extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java [hunk 2]
```diff
@@ -19,6 +20,7 @@
 import io.quarkus.arc.runtime.BeanContainerListener;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.runtime.metrics.MetricsFactory;
```
Comment (kenfinnigan) [inline extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java:23]: Again with added imports

**Review by kenfinnigan — APPROVED** [review]
Couple of notes about imports that don't appear to be used, but otherwise looks good!

**Review by ebullient — APPROVED** [review]

**Pull request merged**